src/types.tsx

```diff
@@ -7,11 +7,29 @@ export type Field<T> = {
   touched: boolean
   error?: string
   rules: ValidationRule<T>[]
-  setValue: (value: T) => void
+  setValue: <
+    U extends SetValueOptions | EmptySetValueOptions = EmptySetValueOptions
```

Comment:
Probably has an obvious answer, but why `SetValueOptions | EmptySetValueOptions = EmptySetValueOptions` instead of `SetValueOptions | EmptySetValueOptions`? I don't understand the syntax here where we appear to be setting a default of `EmptySetValueOptions`.

Reply:
Great question. 

You could certainly drop the default and just do `SetValueOptions | EmptySetValueOptions`, but it would force the caller to explicitly provide the generic when calling `setValue` without the `runValidation` option as TS couldn't infer it `EmptySetValueOptions` from the passed in args.

So `SetValueOptions | EmptySetValueOptions = EmptySetValueOptions` allows the caller to not have to specify the generic.

Reply:
Gotcha—thanks for the explanation!